store/prefixstore.go

```diff
@@ -0,0 +1,59 @@
+package store
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type prefixStore struct {
+	store  KVStore
+	prefix []byte
+}
+
+// Implements Store
+func (s prefixStore) GetStoreType() StoreType {
+	return sdk.StoreTypePrefix
+}
+
+// Implements CacheWrap
+func (s prefixStore) CacheWrap() CacheWrap {
+	return NewCacheKVStore(s)
+}
+
+// Implements KVStore
+func (s prefixStore) Get(key []byte) []byte {
+	return s.store.Get(append(s.prefix, key...))
+}
+
+// Implements KVStore
+func (s prefixStore) Has(key []byte) bool {
+	return s.store.Has(append(s.prefix, key...))
+}
+
+// Implements KVStore
+func (s prefixStore) Set(key, value []byte) {
+	s.store.Set(append(s.prefix, key...), value)
+}
+
+// Implements KVStore
+func (s prefixStore) Delete(key []byte) {
+	s.store.Delete(append(s.prefix, key...))
+}
+
+// Implements KVStore
+func (s prefixStore) Prefix(prefix []byte) KVStore {
+	return prefixStore{s, prefix}
+}
+
+// Implements KVStore
+func (s prefixStore) Iterator(start, end []byte) Iterator {
+	start = append(s.prefix, start...)
```

Comment:
I'm not sure this is the right implementation; it exposes the un-prefixed keys to the iterator caller. Can we create a new `Iterator` wrapper which only exposes the key after the prefix? That keeps the contract `store.Get(iterator.Key()) == iterator.Value()`, which holds for all other store implementations.